redash/query_runner/cass.py

```diff
@@ -0,0 +1,137 @@
+import json
+import sys
+import logging
+
+from redash.query_runner import *
+from redash.utils import JSONEncoder
+
+logger = logging.getLogger(__name__)
+
+class Cassandra(BaseQueryRunner):
+    @classmethod
+    def configuration_schema(cls):
+        return {
+            'type': 'object',
+            'properties': {
+                'host': {
+                    'type': 'string',
+                },
+                'port': {
+                    'type': 'number',
+                    'default': 9042,
+                },
+                'Keyspace': {
+                    'type': 'string',
+                    'title': 'Keyspace name'
+                },
+                'username': {
+                    'type': 'string',
+                    'title': 'Username'
+                },
+                'password': {
+                    'type': 'string',
+                    'title': 'Password'
+                }
+            },
+            'required': ['Keyspace']
+        }
+
+    @classmethod
+    def type(cls):
+        return "Cassandra"
+
+    @classmethod
+    def enabled(cls):
+        try:
+            from cassandra.cluster import Cluster
+        except ImportError:
+            return False
+
+        return True
+
+    def _get_tables(self, schema):
+        query = """
+        select columnfamily_name from system.schema_columnfamilies where keyspace_name = '{}';
+        """.format(self.configuration['Keyspace'])
+
+        results = self.run_query(query)
+        return results, error
+
+    def run_query(self, query):
+        from cassandra.cluster import Cluster
+        connection = None
+        try:
+            if self.configuration.get('username', '') and self.configuration.get('password', ''):
+                auth_provider = PlainTextAuthProvider(username='{}'.format(self.configuration.get('username', '')),
+                                                      password='{}'.format(self.configuration.get('password', '')))
+                connection = Cluster([self.configuration.get('host', '')], auth_provider=auth_provider)
+            else:
+                connection = Cluster([self.configuration.get('host', '')])
+
+            session = connection.connect()
+            logger.info("Cassandra running query: %s", query)
+            result = session.execute(query)
+
+            column_names = []
+            columns = []
+            for i in result.column_names:
+                column_names.append(i)
+
+            for column_name in column_names:
+
+                columns.append({
+                    'name': column_name,
+                    'friendly_name': column_name,
+                    'type': 'string'
+                })
```

Comment:
You can replace this code with the `fetch_columns` method from the `BaseQueryRunner` class.

Something like `columns = self.fetch_columns(map(lambda c: (c, 'string'), column_names))`.